src/monitoring/resourceusageprovider.cpp

```diff
@@ -0,0 +1,376 @@
+/*
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <linux/dqblk_xfs.h>
+#include <sys/quota.h>
+#include <sys/statvfs.h>
+
+#include <filesystem>
+#include <fstream>
+#include <functional>
+#include <numeric>
+#include <regex>
+#include <thread>
+#include <vector>
+
+#include <Poco/StringTokenizer.h>
+
+#include "logger/logmodule.hpp"
+#include "resourceusageprovider.hpp"
+
+namespace aos::sm::monitoring {
+
+namespace {
+
+constexpr std::string_view cCPUTag             = "cpu  ";
```

Comment:
All these consts are too specific to one particular function. Move them inside corresponding function.